src/main.js

```diff
@@ -52,6 +52,33 @@ export async function main(platform) {
         });
         await vm.load();
         platform.createAndMountRootView(vm);
+
+        // needed for mobile Safari which does *not* resize the viewport when the keyboard
+        // shows or hides (see https://bugs.webkit.org/show_bug.cgi?id=141832)
+        if (window.visualViewport) {
```

Comment:
This file shouldn't contain any DOM related code, so I'd move this to `platform.createAndMountRootView`. In there you can see we already add a "legacy" class to the root DOM element if IE11 is detected, I'd do the same for iOS (there's already a check for iOS in `platform/web/DOM/download.js` that you can move to `Platform.js` perhaps and pass that into `downloadInFrame`.

I'd also prefer to only run this code for iOS, so in `createAndMountRootView` you check the iOS flag and conditionally set up this workaround.